modules/apps/portal-template/portal-template-freemarker/src/main/java/com/liferay/portal/template/freemarker/internal/FreeMarkerBundleResourceParser.java

```diff
@@ -27,5 +29,24 @@
 	immediate = true, property = "lang.type=" + TemplateConstants.LANG_TYPE_FTL,
 	service = TemplateResourceParser.class
 )
-public class FreeMarkerBundleResourceParser extends ClassLoaderResourceParser {
+public class FreeMarkerBundleResourceParser extends URLResourceParser {
```

Comment:
@dantewang 
I do not understand why you did this.
After your change, same logic of getURL() will be in three classes: FreemarkerBundlerResoureParser, VelocityBundleResourceParser, and ClassLoaderResourceParser.


Tina.

Reply:
@tinatian It was intended to move ClassLoaderResourceParser to internal package. I'm also thinking of an alternative way to only move part of the ClassLoaderResourceParser logic to internal package and will send another pull.